packages/datetime/test/dateRangeInputTests.tsx

```diff
@@ -487,6 +488,1304 @@ describe("<DateRangeInput>", () => {
+            let endInput: WrappedComponentInput;
+            let getDayElement: (dayNumber?: number, fromLeftMonth?: boolean) => WrappedComponentDayElement;
+            let dayElement: WrappedComponentDayElement;
+
+            beforeEach(() => {
+                const result = wrap(<DateRangeInput defaultValue={[HOVER_TEST_DATE_2, HOVER_TEST_DATE_4]} />);
+
+                root = result.root;
+                getDayElement = result.getDayElement;
+                startInput = getStartInput(root);
+                endInput = getEndInput(root);
+
+                // clear the inputs to start from a fresh state, but do so
+                // *after* opening the popover so that the calendar doesn't
+                // move away from the view we expect for these tests.
+                root.setState({ isOpen: true });
+                changeInputText(startInput, "");
+                changeInputText(endInput, "");
+            });
+
+            function setSelectedRangeForHoverTest(selectedDateConfigs: HoverTextDateConfig[]) {
+                const [startConfig, endConfig] = selectedDateConfigs;
+                changeInputText(startInput, (startConfig == null) ? "" : startConfig.str);
+                changeInputText(endInput, (endConfig == null) ? "" : endConfig.str);
+            }
+
+            describe("when selected date range is [null, null]", () => {
+
+                const SELECTED_RANGE = [null, null];
+                const HOVER_TEST_DATE_CONFIG = HOVER_TEST_DATE_CONFIG_1;
+
+                beforeEach(() => {
+                    setSelectedRangeForHoverTest(SELECTED_RANGE);
+                    dayElement = getDayElement(HOVER_TEST_DATE_CONFIG.day);
+                });
+
+                describe("if start field is focused...", () => {
+
+                    beforeEach(() => {
+                        startInput.simulate("focus");
+                        dayElement.simulate("mouseenter");
+                    });
+
+                    it("shows [<hoveredDate>, null] in input fields", () => {
+                        assertInputTextsEqual(root, HOVER_TEST_DATE_CONFIG.str, "");
+                    });
+
+                    it("keeps focus on start field", () => {
+                        assertStartInputFocused(root);
+                    });
+
+                    describe("on click", () => {
+
+                        beforeEach(() => {
+                            dayElement.simulate("click");
+                        });
+
+                        it("sets selection to [<hoveredDate>, null]", () => {
+                            assertInputTextsEqual(root, HOVER_TEST_DATE_CONFIG.str, "");
+                        });
+
+                        it("moves focus to end field", () => {
+                            assertEndInputFocused(root);
+                        });
+                    });
+
+                    describe("if mouse moves to no longer be over a calendar day...", () => {
```

Comment:
no need for `...`s in describes. we know they're "nested." that's plain unconventional!